AudioDeviceExample/ViewController.swift

```diff
@@ -0,0 +1,466 @@
+
+            previewBounds = previewBounds.integral
+            previewView.bounds = previewBounds
+
+            previewView.center = CGPoint.init(x: view.bounds.width - previewBounds.width / 2 - kPreviewPadding,
+                                              y: view.bounds.height - previewBounds.height / 2 - kPreviewPadding)
+        }
+    }
+
+    override func prefersHomeIndicatorAutoHidden() -> Bool {
+        return self.room != nil
+    }
+
+    override var prefersStatusBarHidden: Bool {
+        return self.room != nil
+    }
+
+    override func willTransition(to newCollection: UITraitCollection, with coordinator: UIViewControllerTransitionCoordinator) {
+        if (newCollection.horizontalSizeClass == .regular ||
+            (newCollection.horizontalSizeClass == .compact && newCollection.verticalSizeClass == .compact)) {
+            remoteViewStack.axis = .horizontal
+        } else {
+            remoteViewStack.axis = .vertical
+        }
+    }
+
+    // Update our UI based upon if we are in a Room or not
+    func showRoomUI(inRoom: Bool) {
+        self.connectButton.isHidden = inRoom
+        self.connectButton.isEnabled = !inRoom
+        self.roomTextField.isHidden = inRoom
+        self.roomLine.isHidden = inRoom
+        self.roomLabel.isHidden = inRoom
+        self.disconnectButton.isHidden = !inRoom
+        self.disconnectButton.isEnabled = inRoom
+        UIApplication.shared.isIdleTimerDisabled = inRoom
+        if #available(iOS 11.0, *) {
+            self.setNeedsUpdateOfHomeIndicatorAutoHidden()
+        }
+        self.setNeedsStatusBarAppearanceUpdate()
+
+        self.navigationController?.setNavigationBarHidden(inRoom, animated: true)
+    }
+
+    func dismissKeyboard() {
+        if (self.roomTextField.isFirstResponder) {
+            self.roomTextField.resignFirstResponder()
+        }
+    }
+
+    func logMessage(messageText: String) {
+        messageLabel.text = messageText
+
+        if (messageLabel.alpha < 1.0) {
+            self.messageLabel.isHidden = false
+            UIView.animate(withDuration: 0.4, animations: {
+                self.messageLabel.alpha = 1.0
+            })
+        }
+
+        // Hide the message with a delay.
+        self.messageTimer?.invalidate()
+        if #available(iOS 10.0, *) {
+            let timer = Timer.init(timeInterval: TimeInterval(6), repeats: false) { (timer) in
+                if (self.messageLabel.isHidden == false) {
+                    UIView.animate(withDuration: 0.6, animations: {
+                        self.messageLabel.alpha = 0
+                    }, completion: { (complete) in
+                        if (complete) {
+                            self.messageLabel.isHidden = true
+                        }
+                    })
+                }
+            }
+
+            self.messageTimer = timer
+            RunLoop.main.add(timer, forMode: .commonModes)
+        } else {
+            // TODO: iOS 9.x support.
+        }
```

Comment:
Another iOS 9 TODO. I may have to use an older `Timer` initializer or deal with the extra complexity of two code paths.